src/io/internal/pipelines/dicom/apply-pstate-to-image.cxx

```diff
@@ -0,0 +1,582 @@
+    unsigned int overlayWidth=0, overlayHeight=0, overlayLeft=0, overlayTop=0;
+    OFBool overlayROI=OFFalse;
+    Uint16 overlayTransp=0;
+    char overlayfile[100];
+    FILE *ofile=NULL;
+
+    max = ps.getNumberOfActiveOverlays(layer);
+
+    // Overlays[]
+    Value overlaysJsonArray(kArrayType);
+
+    for (size_t ovlidx=0; ovlidx<max; ovlidx++)
+    {
+      Value overlayJson(kObjectType);
+      overlayJson.AddMember("Index", Value(static_cast<uint32_t>(ovlidx)), a);
+      std::stringstream value;
+      value << "0x" << STD_NAMESPACE hex << ps.getActiveOverlayGroup(layer, ovlidx) << STD_NAMESPACE dec;
+      overlayJson.AddMember("Group", Value(value.str().c_str(), a), a);
+      overlayJson.AddMember("Label", Value(StringRef(ps.getActiveOverlayLabel(layer, ovlidx))), a);
+      overlayJson.AddMember("Description", Value(StringRef(ps.getActiveOverlayDescription(layer, ovlidx))), a);
+      overlayJson.AddMember("Type", Value(StringRef(ps.activeOverlayIsROI(layer, ovlidx) ? "ROI" : "graphic")), a);
+
+      /* get overlay data */
+      if (EC_Normal == ps.getOverlayData(layer, ovlidx, overlayData, overlayWidth, overlayHeight,
+          overlayLeft, overlayTop, overlayROI, overlayTransp))
+      {
+        overlayJson.AddMember("Width", Value(overlayWidth), a);
+        overlayJson.AddMember("Height", Value(overlayHeight), a);
+        overlayJson.AddMember("Left", Value(overlayLeft), a);
+        overlayJson.AddMember("Top", Value(overlayTop), a);
+
+        std::stringstream buff;
+        buff << "P5\n" << overlayWidth << " " << overlayHeight << " " << "255\n";
+        buff.write((const char*)overlayData, overlayWidth * overlayHeight);
+
+        constexpr bool urlFriendly = false;
+        overlayJson.AddMember("OverlayData", Value(base64_encode(buff.str(), urlFriendly).c_str(), a), a);
+      } else {
+        OFLOG_ERROR(dcmp2pgmLogger, "unable to access overlay data!");
+      }
+      overlaysJsonArray.PushBack(overlayJson, a);
+    }
+    layerJson.AddMember("Overlays", overlaysJsonArray, a); // Overlays[]
+    graphicsLayersJsonArray.PushBack(layerJson, a);
+  }
+  d.AddMember("GraphicsLayers", graphicsLayersJsonArray.Move(), a); // GraphicsLayers[]
+
+  // Pretty print the json into output stream.
+  StringBuffer buffer;
+  PrettyWriter<StringBuffer> writer(buffer);
+  d.Accept(writer);
+  out << buffer.GetString();
+}
+
+
+int main(int argc, char *argv[])
+{
+  itk::wasm::Pipeline pipeline("apply-pstate-to-image", "Apply a presentation state to a given DICOM image and render output as pgm bitmap or dicom file.", argc, argv);
+  // Inputs
+  std::string imageIn;
+  pipeline.add_option("image-in", imageIn, "Input DICOM file")->required()->check(CLI::ExistingFile)->type_name("INPUT_BINARY_FILE");
+
+  // Outputs
+  // Metadata output regarding overlays
+  itk::wasm::OutputTextStream pstateOutStream;
+  pipeline.add_option("pstate-out-stream", pstateOutStream, "Output overlay information")->type_name("OUTPUT_TEXT_STREAM");
```

Comment:
pstate-out-stream -> presentation-state

->type_name("OUTPUT_JSON");

Reply:
@thewtex how can we do the parsing at the javascript output end without using `any` type. Will Typescript and itk-wasm's type checking be fine with it?

Reply:
Yes, we can use `any` for the corresponding Typescript type.

Reply:
I was able to use `Object` as a generic type. Check the latest commit on the branch for changes related to this.